docs/internals/frontends.rst

```diff
@@ -54,11 +88,46 @@ log_message
     created
         Unix timestamp (float)
     levelname
-        Upper-case log level name (also called severity). Defined levels are: DEBUG, INFO, WARNING, CRITICAL
+        Upper-case log level name (also called severity). Defined levels are: DEBUG, INFO, WARNING, ERROR, CRITICAL
     name
         Name of the emitting entity
     message
         Formatted log message
+    :ref:`msgid <msgid>`
+        Message ID, may be *none* or absent
+
+.. rubric:: Examples (reformatted, each object would be on exactly one line)
+
+:ref:`borg_extract` progress::
+
+    {"message": "100.0% Extracting: src/borgbackup.egg-info/entry_points.txt",
+     "current": 13000228, "total": 13004993, "info": ["src/borgbackup.egg-info/entry_points.txt"],
```

Comment:
No, that's what the msgid is for. We _could_ provide the format string, but I think that's less useful, because then the formatting for humans becomes part of the API; ugly. This way we have msgid, which is easy to keep fixed, and *message* which is "don't assume anything" sort of deal: show it verbatim somewhere, or ignore it, but don't try to parse it.